alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,537 @@
+		releaseRequest.ClusterId = d.Id()
+
+		for k, v1 := range newHostGroup {
+			if _, ok := oldHostGroup[k]; ok {
+				newNodeCount, _ := strconv.Atoi(v1["node_count"].(string))
+
+				listHostGroupRequest := emr.CreateListClusterHostGroupRequest()
+				listHostGroupRequest.ClusterId = d.Id()
+				listHostGroupRequest.HostGroupName = k
+
+				raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+					return emrClient.ListClusterHostGroup(listHostGroupRequest)
+				})
+				if err != nil {
+					return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", listHostGroupRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+				}
+
+				resp := raw.(*emr.ListClusterHostGroupResponse)
+				hostGroupId := resp.HostGroupList.HostGroup[0].HostGroupId
+				oldNodeCount := resp.HostGroupList.HostGroup[0].NodeCount
+
+				// scala up
+				if oldNodeCount < newNodeCount {
+					count := newNodeCount - oldNodeCount
+					resizeHostGroup := emr.ResizeClusterV2HostGroup{}
+					resizeHostGroup.ClusterId = d.Id()
+					resizeHostGroup.HostGroupId = hostGroupId
+					resizeHostGroup.NodeCount = strconv.Itoa(count)
+					resizeHostGroup.InstanceType = v1["instance_type"].(string)
+					resizeHostGroup.HostGroupType = v1["host_group_type"].(string)
+
+					resizeHostGroups = append(resizeHostGroups, resizeHostGroup)
+				} else if oldNodeCount > newNodeCount { //scale down
+					releaseRequest.HostGroupId = hostGroupId
+					releaseRequest.InstanceIdList = v1["instance_list"].(string)
+
+					_, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+						return emrClient.ReleaseClusterHostGroup(releaseRequest)
+					})
+
+					if err != nil {
+						return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", releaseRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+					}
+				}
+			}
+		}
+
+		if len(resizeHostGroups) != 0 {
+			resizeRequest.HostGroup = &resizeHostGroups
+
+			_, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+				return emrClient.ResizeClusterV2(resizeRequest)
+			})
+
+			if err != nil {
+				return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_cluster", resizeRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+			}
+		}
+		d.SetPartial("host_group")
+	}
+	d.Partial(false)
+
+	return nil
+}
+
+func resourceAlicloudEmrClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	emrService := EmrService{client}
+	request := emr.CreateReleaseClusterRequest()
+
+	request.Id = d.Id()
+	request.ForceRelease = requests.NewBoolean(true)
+
+	raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+		return emrClient.ReleaseCluster(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
```

Comment:
If releasing a cluster needs a long time, please use self-define timeouts to wait it is deleted. Like https://github.com/terraform-providers/terraform-provider-alicloud/pull/1644/files#diff-e9f12a8f42ad76eaac98a80cadff0dd0R375